archunit/src/test/java/com/tngtech/archunit/testutil/TestUtils.java

```diff
@@ -63,4 +67,21 @@ public int compare(HasName o1, HasName o2) {
             }
         });
     }
+
+    public static URI toUri(URL url) {
+        try {
+            return url.toURI();
+        } catch (URISyntaxException e) {
+            throw new RuntimeException(e);
```

Comment:
I think that introducing the fix here as well might avoid the same problem as the `OutsideOfClassPathRule` uses file system folders. Not completely sure if the caller from `ClassFileImporterTest` is using absolute or relative paths.

Reply:
I think in this case it's not so bad, because the problem only occurs with an OSGI `ClassLoader` AFAIK. But test utils are only used to build ArchUnit itself. And if anybody tries to build ArchUnit using an OSGI `ClassLoader`... Well... :wink: I don't think it's a case we have to guard against.